music21/note.py

```diff
@@ -584,7 +584,7 @@ def _getLyric(self) -> Optional[str]:
             return None
 
         allText = [ly.text for ly in self.lyrics]
-        return '\n'.join(allText)
+        return '\n'.join([t for t in allText if t is not None])
```

Comment:
This solves the crashy part of #1036 but would still be worth a small think over how empty values should roundtrip, i.e. as empty string or None, because right now = inconsistent. Just mentioning in case you're feeling opinionated.

Reply:
(in a future PR)

Reply:
in this case, it's a short-hand so it's not a big deal if it doesn't roundtrip perfectly, since it also doesn't preserve which lyric each syllable (or composite) comes from.